R/methods-epi_archive.R

```diff
@@ -113,26 +113,80 @@ epix_as_of <- function(x, version, min_time_value = -Inf, all_versions = FALSE,
   if (all_versions) {
     # epi_archive is copied into result, so we can modify result directly
     result <- epix_truncate_versions_after(x, version)
-    result$DT <- result$DT[time_value >= .min_time_value, ] # nolint: object_usage_linter
+    if (!identical(.min_time_value, -Inf)) {
+      # See below for why we need this branch.
+      result$DT <- result$DT[time_value >= .min_time_value, ] # nolint: object_usage_linter
+    }
     return(result)
   }
 
   # Make sure to use data.table ways of filtering and selecting
-  as_of_epi_df <- x$DT[time_value >= .min_time_value & version <= .version, ] %>% # nolint: object_usage_linter
-    unique(
-      by = c("geo_value", "time_value", other_keys),
-      fromLast = TRUE
-    ) %>%
-    tibble::as_tibble() %>%
-    dplyr::select(-"version") %>%
-    as_epi_df(
-      as_of = version,
-      other_keys = other_keys
-    )
+  as_of_epi_df <- if (identical(.min_time_value, -Inf)) {
+    # This branch is needed for `epix_as_of` to work with `yearmonth` time type
+    # to avoid time_value > .min_time_value, which is NA for `yearmonth`.
+    x$DT[version <= .version, ]
```

Comment:
Needs removal of the `version` column and conversion to an `epi_df`.  A test is flagging this on my end.

(Aside: this `.version` hack to try to avoid existing colnames actually has a better solution using DT `i` eval rules (`with = FALSE` doesn't work, that's about `j`).  Something like
```r
filter_flags = x$DT$version <= version
x$DT[filter_flags, ]
```
)

Reply:
I fixed the errors, they were coming from some R pipe quirks that I posted about on Slack.

Reply:
Added the DT tweaks too